js/main.js

```diff
@@ -1,5 +1,74 @@
-function clickNext(event) {
+function setGoal(event) {
   event.preventDefault();
+  navSearch();
 }
-var $form = document.querySelector('.form');
-$form.addEventListener('submit', clickNext);
+
+function searchInput(event) {
+  var input = $searchBar.value;
+  if (input.length < 2) {
+    data.results = [];
+    for (var i = 0; i < 4; i++) {
+      $results[i].className = 'result hidden';
+    }
+    return;
+  }
+  var xhr = new XMLHttpRequest();
+  xhr.open('GET', 'https://trackapi.nutritionix.com/v2/search/instant?branded=false&query=' + input);
+  xhr.responseType = 'json';
+  xhr.setRequestHeader('x-app-id', 'c1479c3a');
+  xhr.setRequestHeader('x-app-key', '2f7f3b0e2a3ffe42df018fc46a4cc852');
+  xhr.setRequestHeader('x-remote-user-id', 0);
+  xhr.addEventListener('load', function () {
+    data.results = xhr.response.common;
+    for (var i = 0; i < 4; i++) {
+      $results[i].className = 'result';
+      $imgResults[i].setAttribute('src', data.results[i].photo.thumb);
+      $textResults[i].textContent = data.results[i].food_name;
+    }
+  });
+  xhr.send();
+}
+
+function delayAutofill() {
+  setTimeout(searchInput, 500);
```

Comment:
You should clear the previous timeout before creating a new timeout here. If somebody starts typing again, you want to delay their search for another half second, otherwise you will be sending out a request for each character after 500 milliseconds, no matter what.

Reply:
Ahh okay, yes that makes sense! 😁